roles/openstack/nova/tasks/cleanup.yml

```diff
@@ -0,0 +1,45 @@
+---
+# This file is used to cleanup OpenStack nova resources
+
+- name: Register nova servers details
+  os_server_facts:
+      cloud: "{{ provisioner.cloud | default(omit) }}"
+      server: "{{ prefix }}*"
+  register: servers_details
+- debug: var=servers_details
+
+- name: Set floating IP address
+  set_fact:
+      floating_ip_address: "{{ item.value[0].interface_ip }}"
+  with_dict: servers_details.ansible_facts
+  ignore_errors: yes
```

Comment:
OK so how about:

name: Register nova servers details
to:
name: Cleanup: Set nova servers details

name: Set floating IP address
to:
name: Cleanup - Set floating IP address

name: Release floating IPs
to:
name: Cleanup floating IPs

From ansible documentation "registered variables are just like facts" so we shouldn't distinguish them in the naming of those playbooks.